src/cdk/stepper/stepper.ts

```diff
@@ -349,7 +349,9 @@ export class CdkStepper implements OnDestroy {
     if (this._linear && index >= 0) {
       return steps.slice(0, index).some(step => {
         const control = step.stepControl;
-        const isIncomplete = control ? (control.invalid || control.pending) : !step.completed;
+        const isIncomplete = control ?
+            (control.invalid || control.pending) || !step.interacted :
```

Comment:
You should be able to have the parentheses around the `!step.interacted`. I added the one around the control one so they're easier to distinguish inside the ternary.